src/Microsoft.IdentityModel.Logging/LogHelper.cs

```diff
@@ -375,20 +373,22 @@ private static object SanitizeSecurityArtifact(object arg)
             return arg;
         }
 
-        private static string? RemovePII(object arg)
+        private static string RemovePII(object arg)
         {
             if (arg is Exception ex && IsCustomException(ex))
                 return ex.ToString();
 
             if (arg is NonPII)
-                return arg!.ToString();
+                return arg.ToString()!; // the null check is done in FormatInvariant
 
-            return string.Format(CultureInfo.InvariantCulture, IdentityModelEventSource.HiddenPIIString, arg?.GetType().ToString() ?? "Null");
+            return string.Format(CultureInfo.InvariantCulture, IdentityModelEventSource.HiddenPIIString, arg.GetType().ToString() ?? "Null");
         }
 
         internal static bool IsCustomException(Exception ex)
         {
-            return ex!.GetType().FullName!.StartsWith("Microsoft.IdentityModel.", StringComparison.Ordinal);
+            var fullName = ex.GetType().FullName;
```

Comment:
Although I like this better, this could still throw an ArgumentNullException and we would spend decades finding why. Could we handle the ex being null case? e.g. return string empty or null or throw ANE as appropriate?

Reply:
Would be more helpful to have you review Franco's main PRs for this work.